### PR TITLE
chore: add warning logs for bad runescape settings

### PR DESCRIPTION
**Pull request by iProdigy**
* Collection log notifier requires `COLLECTION_LOG_NOTIFICATION` = 1 or 3
* If `COMBAT_TASK_REPEAT_POPUP` is enabled, warn that combat task will keep sending notifications on each repeated event

### CODE REVIEW

File: src/main/java/dinkplugin/DinkPlugin.java [hunk 1]
```diff
@@ -76,6 +84,37 @@ DinkPluginConfig provideConfig(ConfigManager configManager) {
...
+
+        String key = event.getKey();
+        String value = event.getNewValue();
+        GameState gameState = client.getGameState();
+
+        if ("combatTaskEnabled".equals(key) && "true".equals(value) && gameState == GameState.LOGGED_IN) {
+            clientThread.invokeLater(() -> {
+                if (client.getVarbitValue(Varbits.COMBAT_ACHIEVEMENTS_POPUP) < 1) {
+                    log.warn("Combat Achievements popup is not enabled in RuneScape settings; Dink Combat Task notifier will not fire.");
+                }
```
Comment (Felanbird) [inline src/main/java/dinkplugin/DinkPlugin.java:100]: `COMBAT_ACHIEVEMENT_POPUP` is not synonymous with combat achievements as a whole being enabled, from what I can see there's no way to disable them.

Collection Log has a popup setting, but it is separate from the write-to-chat setting.

**Review by Felanbird — APPROVED** [review]
image

File: src/main/java/dinkplugin/DinkPlugin.java [hunk 2]
```diff
@@ -43,6 +52,15 @@
 )
 public class DinkPlugin extends Plugin {
 
+    private static final Color PINK = ColorUtil.fromHex("#eba2c0");
+    private static final Color RED = ColorUtil.fromHex("#f09d9e");
```
Comment (iProdigy) [inline src/main/java/dinkplugin/DinkPlugin.java:56]: Open to a different red (this was chosen from analogous on color wheel)

image

File: src/main/java/dinkplugin/DinkPlugin.java [hunk 3]
```diff
@@ -76,6 +94,33 @@ DinkPluginConfig provideConfig(ConfigManager configManager) {
         return configManager.getConfig(DinkPluginConfig.class);
     }
 
+    @Subscribe
+    public void onConfigChanged(ConfigChanged event) {
```
Comment (iProdigy) [inline src/main/java/dinkplugin/DinkPlugin.java:98]: should this logic be moved to a separate class

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,7 @@
 ## Unreleased
 
+- Dev: Add warning logs for improper runescape settings that impact notifiers. (#92)
```
Comment (pajlada) [inline CHANGELOG.md:3]: Should probably be changed to the `Minor` category now that it has  user-visible impact

File: src/main/java/dinkplugin/DinkPlugin.java [hunk 5]
```diff
@@ -143,6 +188,14 @@ public void onLootReceived(LootReceived lootReceived) {
     @Subscribe
     public void onVarbitChanged(VarbitChanged event) {
         diaryNotifier.onVarbitChanged(event);
+
+        if (event.getVarbitId() == CombatTaskNotifier.COMBAT_TASK_REPEAT_POPUP && event.getValue() > 0 && config.notifyCombatTask()) {
+            log.warn("Repeat popups for Combat Achievements is enabled; Dink Combat Task notifier will repeatedly fire on each completion.");
```
Comment (pajlada) [inline src/main/java/dinkplugin/DinkPlugin.java:193]: Do we keep these as `log.warn`?
I imagine if they're changed to send a chat message it'd be spammy on world hop/login but worth considering, but maybe for a separate PR once this has been in a stable release

**Review by pajlada — APPROVED** [review]
Minor nitpicks/questions, but overall looks good. I like the chat message and the look of it.

File: src/main/java/dinkplugin/DinkPlugin.java (hunk 5 above)
Comment (iProdigy) [reply src/main/java/dinkplugin/DinkPlugin.java:193]: Yeah, went for `warn` as a decent middleground between `debug` (won't appear in client.log without `--debug`) and chat (would be spammy if the setting was actually desired) - but can be revised later too

File: src/main/java/dinkplugin/DinkPlugin.java (hunk 2 above)
Comment (Felanbird) [reply src/main/java/dinkplugin/DinkPlugin.java:56]: Not a readable color on default chat background
image

Comment (iProdigy) [reply src/main/java/dinkplugin/DinkPlugin.java:56]: Good catch! will test alternatives later tonight

**Comment by Felanbird** [comment]
I believe the collection log notif that fires when logging/world hopping is using backwards logic. I'm getting a notif when logging in, and the setting is enabled, but not when it's disabled.

File: src/main/java/dinkplugin/DinkPlugin.java (hunk 2 above)
Comment (iProdigy) [reply src/main/java/dinkplugin/DinkPlugin.java]: Latest colors
image

File: src/main/java/dinkplugin/DinkPlugin.java (hunk 5 above)
Comment (iProdigy) [reply src/main/java/dinkplugin/DinkPlugin.java]: Improved in https://github.com/pajlads/DinkPlugin/pull/92/commits/ae2aaba18dcb98e1e74efc6e4418d80f5bb707c4

**Comment by pajlada** [comment]
This color choice looks fine to me

opaque
transparent

**Comment by Felanbird** [comment]
I only tested the "will it break?" aspect of the most recent changes, and I did not run into said issue.

File: src/main/java/dinkplugin/DinkPlugin.java (hunk 3 above)
Comment (pajlada) [reply src/main/java/dinkplugin/DinkPlugin.java:99]: I think this is very much up to you if it makes sense from a Java perspective to have a specific class be responsible for this. I could see a "SettingValidator" class or something have the responsibility for checking & posting warnings, while this class would just post events into it but up to you

**Review by pajlada — APPROVED** [review]
I'm happy with this change as-is, you can move the stuff from DinkPlugin,java into its own class if you feel like it makes sense

The error that Felanbird has mentioned is a bit difficult to track down, I imagine it will not be reproducible in a real build or by real human beings

**Review by Mm2PL — CHANGES_REQUESTED** [review]
I like the feature but i have a nit pick regarding checking varbit values. Current chat colors are fine imo

**Review by Mm2PL — APPROVED** [review]

**Pull request merged**